Candidate: CVE-2024-37891
PublicDate: 2024-06-17 20:15:00 UTC
References:
 https://www.cve.org/CVERecord?id=CVE-2024-37891
 https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf
Description:
 urllib3 is a user-friendly HTTP client library for Python. When using
 urllib3's proxy support with `ProxyManager`, the `Proxy-Authorization`
 header is only sent to the configured proxy, as expected. However, when
 sending HTTP requests *without* using urllib3's proxy support, it's
 possible to accidentally configure the `Proxy-Authorization` header even
 though it won't have any effect as the request is not using a forwarding
 proxy or a tunneling proxy. In those cases, urllib3 doesn't treat the
 `Proxy-Authorization` HTTP header as one carrying authentication material
 and thus doesn't strip the header on cross-origin redirects. Because this
 is a highly unlikely scenario, we believe the severity of this
 vulnerability is low for almost all users. Out of an abundance of caution
 urllib3 will automatically strip the `Proxy-Authorization` header during
 cross-origin redirects to avoid the small chance that users are doing this
 on accident. Users should use urllib3's proxy support or disable automatic
 redirects to achieve safe processing of the `Proxy-Authorization` header,
 but we still decided to strip the header by default in order to further
 protect users who aren't using the correct approach. We believe the number
 of usages affected by this advisory is low. It requires all of the
 following to be true to be exploited: 1. Setting the `Proxy-Authorization`
 header without using urllib3's built-in proxy support. 2. Not disabling
 HTTP redirects. 3. Either not using an HTTPS origin server or for the proxy
 or target origin to redirect to a malicious origin. Users are advised to
 update to either version 1.26.19 or version 2.2.2. Users unable to upgrade
 may use the `Proxy-Authorization` header with urllib3's `ProxyManager`,
 disable HTTP redirects using `redirects=False` when sending requests, or
 not user the `Proxy-Authorization` header as mitigations.
Ubuntu-Description:
Notes:
 mdeslaur> On focal and earlier, the python-pip package bundles
 mdeslaur> python-urllib3 binaries when built. After updating
 mdeslaur> python-urllib3, a no-change rebuild of python-pip is required.
 mdeslaur> On jammy and later, python-urllib3 is bundled in the python-pip
 mdeslaur> package and needs to be patched.
Mitigation:
Bugs:
Priority: low
 Per urllib3 developers, this is a highly unlikely scenario
Discovered-by:
Assigned-to: hlibk
CVSS:

Patches_python-urllib3:
 upstream: https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e (2.x)
 upstream: https://github.com/urllib3/urllib3/commit/40b6d1605814dd1db0a46e202d6e56f2e4c9a468 (1.26.x)
upstream_python-urllib3: released (2.2.2, 1.26.19)
trusty/esm_python-urllib3: needs-triage
esm-infra-legacy/trusty_python-urllib3: needs-triage
esm-infra/xenial_python-urllib3: needed
esm-infra/bionic_python-urllib3: needed
focal_python-urllib3: needed
jammy_python-urllib3: needed
mantic_python-urllib3: ignored (end of life, was needs-triage)
noble_python-urllib3: needed
oracular_python-urllib3: needed
devel_python-urllib3: needs-triage

Patches_python-pip:
upstream_python-pip: needs-triage
trusty/esm_python-pip: needs-triage
esm-infra-legacy/trusty_python-pip: needs-triage
esm-apps/xenial_python-pip: needed
esm-apps/bionic_python-pip: needed
focal_python-pip: needed
esm-apps/focal_python-pip: needed
jammy_python-pip: needed
esm-apps/jammy_python-pip: needed
mantic_python-pip: ignored (end of life, was needs-triage)
noble_python-pip: needed
esm-apps/noble_python-pip: needed
oracular_python-pip: needed
devel_python-pip: needs-triage
